driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -81,11 +79,12 @@
     private volatile boolean isClosed;
     private volatile ClusterDescription description;
 
-    BaseCluster(final ClusterId clusterId, final ClusterSettings settings, final ClusterableServerFactory serverFactory) {
+    BaseCluster(final ClusterId clusterId, final ClusterSettings settings, final ClusterListener clusterListener,
```

Comment:
The bulk of the changes are due to the fact that we can no longer get the listener directly from the settings.  It has to be wrapped at a higher level.